### PR TITLE
fixes #1101: Test Export CSV Enterprise version

### LINKED ISSUE #1101: Test Export CSV Enterprise version
Provide tests in v 3.5 for Export CSV Enterprise features (NODE KEY constraints)

### PR DESCRIPTION
**Pull request by conker84**
Fixes #1101 

Added the `ExportCypherEnterpriseFeaturesTest` for specific enterprise features tests. On Travis we get FileNotFoundException with "Permission Denied" so we test "stream" feature using the result map in orde to assert the statement

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  - Added the `ExportCypherEnterpriseFeaturesTest` for specific enterprise features tests

### CODE REVIEW

File: src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                (r) -> assertExportStatement(EXPECTED_NEO4J_SHELL_WITH_COMPOUND_CONSTRAINT, output, r));
+    }
+
+    private void assertExportStatement(String expectedStatement, File output, Map<String, Object> result) {
+        assertEquals(expectedStatement, isTravis() ? result.get("cypherStatements") : readFile(output));
+    }
+
+    private static String readFile(File output) {
+        try {
+            return new Scanner(output).useDelimiter("\\Z").next() + String.format("%n");
```
Comment (jexp) [inline src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java]: we should move those into utilities or use it from there.
also in Java8 we should probably rather use something from java.nio.Files https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#lines-java.nio.file.Path-

File: src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String getFilePath(File output) { // TODO on Travis we get FileNotFoundException with "Permission Denied" so we test the Cypher from the map (investigate why)
```
Comment (jexp) [inline src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java]: should this go into testutils too?

File: src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        return isTravis() ? "null" : PREFIX + output.getPath().replace("target/", "");
+    }
+
+    private File getFile(String fileName) {
```
Comment (jexp) [inline src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java]: what is this achieving?
the method name is not really clear

it actually writes a  new, empty file to the location?

What about file.exits(), canWrite(), isDirectory() etc?

Comment (conker84) [reply src/test/java/apoc/export/cypher/ExportCypherEnterpriseFeaturesTest.java]: They are tries that I made when I realized that on Travis I had permissions issue, I forgot to delete them sorry.

**Comment by conker84** [comment]
I cleaned/moved the code a requested, the only exceptions are `getFilePath` and `getImportDirectoryPath` which are test specific.

**Pull request merged**